app/models/target.rb

```diff
@@ -30,12 +30,43 @@ class Target < ApplicationRecord
 
   validate :limit_number_of_targets
 
+  after_create :check_matching_targets
+
+  reverse_geocoded_by :latitude, :longitude
+
+  scope :match_not_user_topic, lambda { |params|
+    where(topic_id: params[0]).where.not(user_id: params[1])
+  }
+
   private
 
   def limit_number_of_targets
     return if user && user.targets.count < 10
 
-    errors.add(:base, I18n
-      .t('api.error.invalid_request.max_target_limit'))
+    errors.add(:base,
+               I18n.t('api.error.invalid_request.max_target_limit'))
+  end
+
+  def check_matching_targets
```

Comment:
I didn't see tests for this method and it's an important one. Maybe are in another PR?

Reply:
You can find this tests in the next PR.